### PR TITLE
Stormdrive Update 2

### LINKED ISSUE #33: Reactor Startup Procedure
The reactor startup procedure is sort of janky and not totally clear about what needs to happen in order to turn the reactor on. It also lacks basic troubleshooting help.

### LINKED ISSUE #158: Constrictor boards don't exist
The hammerhead has no plasma constrictor boards in tech storage, rendering them irreplaceable should an incident destroy the constrictors. Tech storage should have at least two, as it is likely that if one breaks, the other will as well.

### LINKED ISSUE #210: Stormdrive doesn't free reference if parent gas pump is deleted
## Reproduction:
Unwrench or otherwise qdel the reactor manifold. GC has to queue a harddel

### LINKED ISSUE #273: Access restriction problem with reactor console
## Round ID:
70

## Reproduction:
Be an atmos tech (or possibly any role, I dunno)
Try to use the nuclear reactor console
Get "Access denied" warning in chat but have screen come up and be able to control it anyways

### PR DESCRIPTION
**Pull request by Karmic-Skink**
## About The Pull Request
This PR reworks magnetic constrictors to be atmos machines, dealing with the maxing pumps issue. Also a few extra stormdrive related things.

## Why It's Good For The Game
Stormdrive working as intended is good for game.
fixes #273
fixes #210
fixes #158
fixes #33

## Changelog
:cl:
add: Added magnetic constrictor machine boards
add: Added magnetic constrictor research path
add: Added reactor control console computer boards
add: Added reactor control console research path
add: Added additional stormdrive behaviours
add: Added reactor manifold pump
fix: Fixed reactor control console access bug
imageadd: Added additional magnetic constrictor sprites 
balance: Changed access requirements on engine crates
refactor: Reworked Magnetic Constrictors to be Atmos Machines
admin: Added an admin proc for directly adding fuel to the stormdrive
/:cl:

### CODE REVIEW

File: _maps/map_files/Hammerhead/Hammerhead.dmm [hunk 1]
```diff
@@ -75,7 +75,8 @@
 /area/engine/storage)
 "aan" = (
 /obj/machinery/atmospherics/pipe/simple/purple/visible{
-	dir = 10
+	dir = 10;
+	step_y = 0
```
Comment (Vasily2013) [inline _maps/map_files/Hammerhead/Hammerhead.dmm]: Remove this line

Comment (Karmic-Skink) [reply _maps/map_files/Hammerhead/Hammerhead.dmm]: Done.

File: code/modules/cargo/packs.dm [hunk 2]
```diff
@@ -888,6 +888,7 @@
 
 /datum/supply_pack/engine
 	group = "Engine Construction"
+	access = ACCESS_ENGINE_EQUIP //NSV13 - Do your hecking jobs
```
Comment (Vasily2013) [inline code/modules/cargo/packs.dm]: nice

File: nsv13/code/modules/power/stormdrive.dm [hunk 3]
```diff
@@ -326,6 +188,11 @@ Takes  plasma and outputs superheated plasma and a shitload of radiation.
 	air1.gases[/datum/gas/constricted_plasma][MOLES] += 300
 	try_start()
 
+/obj/machinery/power/stormdrive_reactor/proc/juice_up(var/juice) //Admin command to add a specified amount of CPlas to the drive
```
Comment (Vasily2013) [inline nsv13/code/modules/power/stormdrive.dm]: this gon be nice.

File: _maps/map_files/Hammerhead/Hammerhead.dmm [hunk 4]
```diff
@@ -21328,14 +21311,10 @@
...
-	desc = "A pump that moves gas by pressure. <b>Some moron's taken a hacksaw to it to act as the Stormdrive's gas manifold. Best to leave it alone.</b>";
-	dir = 8;
-	name = "Reactor Manifold"
-	},
 /obj/effect/landmark/event_spawn,
-/obj/structure/cable/yellow{
-	icon_state = "0-4"
+/obj/machinery/atmospherics/components/binary/pump/reactor_manifold{
+	dir = 8;
+	step_x = 0
```
Comment (Vasily2013) [inline _maps/map_files/Hammerhead/Hammerhead.dmm]: There's another here.

Comment (Karmic-Skink) [reply _maps/map_files/Hammerhead/Hammerhead.dmm]: Someone needs to disable that option.

File: nsv13/code/modules/power/stormdrive.dm [hunk 5]
```diff
@@ -439,13 +319,36 @@ Takes  plasma and outputs superheated plasma and a shitload of radiation.
...
+	for(var/obj/O in orange(6, src))
+		if(!O.anchored)
+			step_towards(O,src)
+	for(var/mob/living/M in orange(6, src))
+		if(!M.mob_negates_gravity())
+			step_towards(M,src)
+			M.Knockdown(40) //Knockdown prey so it can't get away!
+
+/obj/machinery/power/stormdrive_reactor/proc/ambient_temp_bleed()
+	var/turf/open/L = loc
```
Comment (Kmc2000) [inline nsv13/code/modules/power/stormdrive.dm]:
```suggestion
	var/turf/open/L = get_turf(src)
```
Is the standard

File: nsv13/code/modules/power/stormdrive.dm [hunk 6]
```diff
@@ -439,13 +319,36 @@ Takes  plasma and outputs superheated plasma and a shitload of radiation.
...
+	if(!istype(L) || !(L.air))
+		return
+	var/datum/gas_mixture/env = L.return_air()
+//	var/heat_capacity = env.heat_capacity()
```
Comment (Kmc2000) [inline nsv13/code/modules/power/stormdrive.dm]:
```suggestion
```

File: nsv13/code/modules/power/stormdrive.dm [hunk 7]
```diff
@@ -439,13 +319,36 @@ Takes  plasma and outputs superheated plasma and a shitload of radiation.
...
+		env.temperature += delta_env / 2
+		air_update_turf()
 
 /obj/machinery/power/stormdrive_reactor/proc/can_cool()
 	if(heat > REACTOR_HEAT_NORMAL+10) //Only start melting the rods if theyre running it hot. We have a "safe" mode which doesn't need you to check in on the reactor at all.
 		rod_integrity -= input_power_modifier/60 //Assuming youre running it at hot, rods will melt every 30 minutes.
 		if(rod_integrity < 0)
 			rod_integrity = 0
 			send_alert("DANGER: Primary control rods have failed!")
+			message_admins("Stormdrive control rods are at 0% integrity.")
```
Comment (Kmc2000) [inline nsv13/code/modules/power/stormdrive.dm]: Do we not want players to know that control rods have failed?

File: nsv13/code/modules/power/stormdrive.dm [hunk 8]
```diff
@@ -474,53 +377,280 @@ Takes  plasma and outputs superheated plasma and a shitload of radiation.
...
+	design_ids = list("sd_r_c_c")
+	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
+	export_price = 5000
+
+//////Reactor Manifold//////
+
+/obj/machinery/atmospherics/components/binary/pump/reactor_manifold
+	name = "Reactor Manifold"
+	desc = "The manifold connection for the stormdrive."
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
```
Comment (Kmc2000) [inline nsv13/code/modules/power/stormdrive.dm]: I still hate myself for making this monstrosity but it'll do for the time being 🤣

File: nsv13/code/modules/power/stormdrive.dm (hunk 7 above)
Comment (Karmic-Skink) [reply nsv13/code/modules/power/stormdrive.dm]: They already do get notified through send_alert().
This is just a direct to admins message.

File: nsv13/code/modules/power/stormdrive.dm (hunk 8 above)
Comment (Karmic-Skink) [reply nsv13/code/modules/power/stormdrive.dm]: It is a time effective solution to the only issue I've seen arise from it using this method so far.
Rewriting it all is not time effective given the limited resources at our disposal.

File: nsv13/code/modules/power/stormdrive.dm (hunk 5 above)
Comment (Karmic-Skink) [reply nsv13/code/modules/power/stormdrive.dm]: Done.

File: nsv13/code/modules/power/stormdrive.dm (hunk 6 above)
Comment (Karmic-Skink) [reply nsv13/code/modules/power/stormdrive.dm]: Done.

**Pull request merged**